Mathlib/Topology/Sober.lean

```diff
@@ -112,6 +113,11 @@ theorem IsIrreducible.genericPoint_spec [QuasiSober α] {S : Set α} (hS : IsIrr
     IsGenericPoint hS.genericPoint (closure S) :=
   (QuasiSober.sober hS.closure isClosed_closure).choose_spec
 
+theorem IsIrreducible.genericPoint_spec' [QuasiSober α] {S : Set α}
+    (hS : IsIrreducible S) (hS' : IsClosed S) :
+    IsGenericPoint hS.genericPoint S := by
+  convert hS.genericPoint_spec; exact hS'.closure_eq.symm
```

Comment:
The following might be out of scope of this PR, but to me it feels a bit odd that this version has the `'`, and that it is not immediate by definition, given how `QuasiSober` is defined (that closed irr sets have generic points).

I understand that it is useful for `IsIrreducible.genericPoint` to make sense for non-closed sets as well, but then shouldn't the definition of `QuasiSober` already reflect that? So that its defining property would instead be
` sober : ∀ {S : Set α}, IsIrreducible S → ∃ x, IsGenericPoint x (closure S)`
Then it would make more sense to me that the prime versions are for the case of closed sets.

Reply:
I changed the names a bit. Please check if you like them.